### ISSUE TITLE
Remove Ruby/nanoc dependency

### ISSUE THREAD

**Issue by ajpiano** [jquery/learn.jquery.com#120]
Nanoc has taken us pretty far with this site, but the frankenbuild of using node.js to shell out to nanoc (requiring a functioning ruby environment) is hairy and adds a superfluous dependency. The building of the markdown content into html can be handled by a node-based static site generator like [Wintersmith](https://github.com/jnordberg/wintersmith) instead, and the custom logic in our Rules file can also be ported there as well. 

TLDR: requiring node, php, ruby, and python to build the site fully locally feels insane (and inconvenient)
